test/math.jl

```diff
@@ -387,12 +387,20 @@ j33 = besselj(3,3.)
 @test besselj(-3,-3) == j33
 @test besselj(-3,3) == -j33
 @test besselj(3,-3) == -j33
+@test besselj(3,3f0) ≈ j33
```

Comment:
but if we round the higher-precision value into the lower precision type can we reliably test exact equality?